src/nspawn/nspawn.c

```diff
@@ -3206,6 +3206,17 @@ static int outer_child(
         return 0;
 }
 
+static int on_sigchld(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
+        pid_t pid;
+
+        pid = PTR_TO_UINT32(userdata);
+        if (pid == si->ssi_pid) {
```

Comment:
How can I get `pid != si->ssi_pid` here?
Yes, I've read:
```
as it may also be coming
from an orphan process
```
but it would be great to see the real example.